### ISSUE TITLE
Certain nesting of ARIA attributes causes NVDA to ignore active elements in the tab order in IE

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3571]
**Reported by bgaraventa on 2013-10-10 20:45**
It appears that, when certain ARIA attributes are nested, such as role=dialog and role=application, NVDA cannot read active elements in the tab order when viewed in IE, though it does work correctly in FF.

Here is an example:
```
<html><body>

<div>
Content...
</div>

<div role="dialog">
<table role="application">
<tr>
<td role="link" tabindex="0" aria-label="Test Link">
?
</td>
</tr>
</table>
</div>

</body></html>
```
When you attempt to press Tab to set focus to the "Test Link", it does work correctly in FF and reads both the role and state accurately.

However if you try the same in IE, it causes a dead tab stop and appears to get stuck so you can't tab past it.

I understand that this is a corner case, but the active element shouldn't be entirely invisible in the tab order regardless.

**Comment by Adriani90** [comment]
@bgaraventa I am not sure if I can reproduce this issue in IE11. I am testing with NVDA las Alpha and am getting following error. As you can see, I could tab to the link and then to "show options" and the focus didn't got stuck on that region. But I am not sure if there are things ignored. Can you reproduce this issue with last NVDA Alpha?


```
IO - inputCore.InputManager.executeGesture (18:39:16.891):
Input: kb(laptop):NVDA+space
IO - inputCore.InputManager.executeGesture (18:39:18.430):
Input: kb(laptop):tab
ERROR - core.CorePump.run (18:39:18.532):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 486, in run
  File "IAccessibleHandler.pyc", line 900, in pumpAll
  File "IAccessibleHandler.pyc", line 671, in processFocusWinEvent
  File "IAccessibleHandler.pyc", line 687, in processFocusNVDAEvent
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 640, in _get_shouldAllowIAccessibleFocusEvent
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyc", line 666, in _get_shouldAllowIAccessibleFocusEvent
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 793, in _get_states
  File "comtypes\client\lazybind.pyc", line 226, in __call__
  File "comtypes\automation.pyc", line 729, in _invoke
COMError: (-2147352319, None, (None, None, None, 0, None))
IO - inputCore.InputManager.executeGesture (18:39:19.151):
Input: kb(laptop):tab
IO - speech.speak (18:39:19.331):
Speaking [u'Link']
IO - inputCore.InputManager.executeGesture (18:39:19.760):
Input: kb(laptop):tab
IO - speech.speak (18:39:20.000):
Speaking [u'Show options  Schalter  Reduziert  Untermen\xfc']
```

**Comment by Adriani90** [comment]
I am closing this one as abandoned. We cannot reach to the initial author and the case is very corner. Actually I don't know exactly what is expected here. Testing with NVDA 2019.1.1, NVDA reads the dialog and application role when tabbing in focus mode in IE11. If someone has further thoughts on this, please comment on it and we can reopen it. Thanks.
